proto/ibc/core/packetserver/v1/counterparty.proto

```diff
@@ -0,0 +1,22 @@
+syntax = "proto3";
+
+package ibc.core.packetserver.v1;
+
+option go_package = "github.com/cosmos/ibc-go/v9/modules/core/packet-server/types";
+
+import "gogoproto/gogo.proto";
+import "ibc/core/commitment/v2/commitment.proto";
+
+// Counterparty defines the counterparty for a light client to implement IBC eureka protocol
+// this value is stored under our side's channel ID. which we will use to write all packet messages
+// sent to counterparty
+message Counterparty {
+  // the client identifier of the counterparty chain
+  // client id of the counterparty stored on our chain
+  string client_id = 1;
```

Comment:
I think this can be removed now. I see this mapping as:

key: my identifier (v1: channel/port, v2: clientid)
value: counterparty id, merkle path

The client id used to verify the counterparty is in the key, or can be resolved in the v1 case. It seems redundant atm, I would guess in the actual implementation it would only be referenced by the grpc, but we should probably create a new struct for grpc, like we did with the other types `IdentifiedClientState` for example

Reply:
really we have a mapping from: our reserved key path -> counterparty reserved key path

resolving the client id responsible for the key path, should happen outside of this mapping?

Reply:
Before I forget, I went down the thought rabbit hole on this. The proposed construction works perfectly fine, but there's just some minor considerations one could have:

The current construction embeds the clientID into the `Counterparty`. When we handle v1 cases, the clientID is resolved and set into the counterparty type. In the v2 case, the value is redundant as it is in the key setting the value. The main benefit is code structure as it bundles all the necessary components into a single type. One could also return the clientID in the get counterparty function to achieve the same functionality. 

The main reason to keep this approach, would be for future scenarios where we delete the channel/connection mappings and you want to migrate such that all v1 mappings are directly embedded into the counterparty type. One could have two mappings, as I hint at in the above comment. One to obtain the counterparties keyspace which is set for all v1/v2 and one to obtain the light client either directly in v2 case, or via a lookup in v1 case. 

The reason why I went this rabbit hole was primarily due to naming. `counterparty.clientID` referring to your local client identifier, and not a client identifier on the counterparty, could be confusing. Alternatively, if you renamed the `Counterparty` to something like `Connection` or `Channel`, then `connection.clientId` and `connection.counterpartyID` could make sense. Either way, just a minor consideration.